dsl_parser/tests/test_functions.py

```diff
@@ -718,6 +718,57 @@ def get_property_from_get_property(self):
         plan = prepare_deployment_plan(self.parse(yaml))
         self.assertEqual('secret', plan['nodes'][0]['properties']['b'])
 
+    def test_get_property_not_eager(self):
+        yaml = """
+node_types:
+    vm_type:
+        properties:
+            a: { type: string }
+            c: { type: string }
+node_templates:
+    vm1:
+        type: vm_type
+        properties:
+            a: {get_property: [SELF, c]}
+            c: secret
+"""
+        plan = prepare_deployment_plan(self.parse(yaml), functions_eager=False)
+        self.assertEqual(plan['nodes'][0]['properties']['a'],
+                         {'get_property': ['SELF', 'c']})
+        node = functions.evaluate_node_functions(plan['nodes'][0], None)
+        self.assertEqual(node['properties']['a'], 'secret')
+
+    def test_get_property_another_node_not_eager(self):
+        yaml = """
+node_types:
+    vm_type:
+        properties:
+            a: { type: string }
+            c: { type: string }
+node_templates:
+    vm1:
+        type: vm_type
+        properties:
+            a: {get_property: [vm2, c]}
+            c: secret1
+    vm2:
+        type: vm_type
+        properties:
+            a: xxx
+            c: secret2
+"""
+        plan = prepare_deployment_plan(self.parse(yaml), functions_eager=False)
```

Comment:
and eager tests that fail with runtime ?

Reply:
ditto previous comment, re-added/changed a bit preexisting tests